### PR TITLE
Metamask Upgrade Changes

### PR DESCRIPTION
**Pull request by ksridharbabuus**
Metamask Upgrade Changes

### CODE REVIEW

File: src/utility/BlockchainHelper.js [hunk 1]
```diff
@@ -370,3 +430,136 @@ const getRFAIContractAddress = () => {
...
+};
+
+// Fetching The RFAI Token Balance
+export const getRFAITokenBalance = async metamaskDetails => {
+  let rfaiTokenBalance = 0;
+  const rfaiContractAddress = getRFAIContractAddress();
+  const accountAddress = metamaskDetails.account;
+
+  if (metamaskDetails.isTxnsAllowed) {
+    const web3 = new Web3(process.env.REACT_APP_INFURA_ENDPOINT);
```
Comment (Vivek205) [inline src/utility/BlockchainHelper.js:497]: Web3 is initialised in many methods in this file. Why can’t we create a class and assign web3 to a property ? @sridhar

Comment (Vivek205) [inline src/utility/BlockchainHelper.js:497]: Web3 is initialised in many methods in this file. Why can’t we create a class and assign web3 to a property ? @sridhar

**Review by Vivek205 — APPROVED** [review]

File: src/utility/BlockchainHelper.js (hunk 1 above)
Comment (ksridharbabuus) [reply src/utility/BlockchainHelper.js:497]: The reason is if the metamask is closed by the user the web3 will become invalid. We need to recreate again. Instead on demand creating the object to consider the current instance of metamask.

Comment (ksridharbabuus) [reply src/utility/BlockchainHelper.js:497]: The reason is if the metamask is closed by the user the web3 will become invalid. We need to recreate again. Instead on demand creating the object to consider the current instance of metamask.

**Pull request merged**